src/main/scala/viper/server/core/ViperCoreServer.scala

```diff
@@ -0,0 +1,79 @@
+package viper.server.core
+
+import akka.actor.ActorRef
+import viper.server.ViperConfig
+import viper.server.core.ViperBackendConfigs._
+import viper.server.vsi.{Letter, ProcessManagement, VerificationJobHandler}
+import viper.silver.ast
+import viper.silver.logger.ViperLogger
+
+import scala.language.postfixOps
+
+class SilverLetter() extends Letter
+
+class ViperCoreServer(private var _config: ViperConfig) extends ProcessManagement {
+
+  // --- VCS : Configuration ---
+
+  final def config: ViperConfig = _config
+
+  protected var _logger: ViperLogger = _
+  final def logger: ViperLogger = _logger
+
+
+  /** Configures an instance of ViperCoreServer.
+    *
+    * This function should be called before any other.
+    * */
+  def start(): Unit = {
+    println("Hi from VCS start")
+    config.verify()
+
+    _logger = ViperLogger("ViperServerLogger", config.getLogFileWithGuarantee, config.logLevel())
+    println(s"Writing [level:${config.logLevel()}] logs into ${if (!config.logFile.isSupplied) "(default) " else ""}journal: ${logger.file.get}")
+
+    ViperCache.initialize(logger.get, config.backendSpecificCache())
+
+    println(s"ViperServer online in CoreServer mode")
+    super.start(config.maximumActiveJobs())
+  }
+
+  /** Verifies a Viper AST using the specified backend.
+    * */
+  def verify(programID: String, backend_config: ViperBackendConfig, program: ast.Program): VerificationJobHandler = {
+    val args: List[String] = backend_config match {
+      case _ : SiliconConfig => "silicon" :: backend_config.partialCommandLine
+      case _ : CarbonConfig => "carbon" :: backend_config.partialCommandLine
+      case _ : CustomConfig => "DummyFrontend" :: backend_config.partialCommandLine
```

Comment:
Why is the name of the custom verifier "DummyFrontend"? It should be parametric. E.g., in case one implements an extension of Carbon, the name could be "carbon-with-advanced-features".

Reply:
Maybe an instance of ```CustomConfig``` should know what the name is.